src/cursor/cur_backup.c

```diff
@@ -359,18 +359,18 @@ __backup_start(WT_SESSION_IMPL *session,
 	}
 
 err:	/* Close the hot backup file. */
-	if (cb->bfs != NULL)
-		WT_TRET(__wt_fclose(session, &cb->bfs));
 	if (srcfs != NULL)
 		WT_TRET(__wt_fclose(session, &srcfs));
 	if (ret == 0) {
 		WT_ASSERT(session, dest != NULL);
-		WT_TRET(__wt_fs_rename(session, WT_BACKUP_TMP, dest, false));
+		WT_TRET(__wt_sync_and_rename(session,
+		    &cb->bfs, WT_BACKUP_TMP, dest));
 		__wt_writelock(session, &conn->hot_backup_lock);
 		conn->hot_backup_list = cb->list;
 		__wt_writeunlock(session, &conn->hot_backup_lock);
 		F_SET(session, WT_SESSION_BACKUP_CURSOR);
-	}
+	} else if (cb->bfs != NULL)
+		WT_TRET(__wt_fclose(session, &cb->bfs));
```

Comment:
I would do this as:
```
--- cur_backup.c        2019-03-04 10:30:08.246209000 -0500
+++ -   2019-03-04 10:35:11.202636339 -0500
@@ -369,7 +369,8 @@
                conn->hot_backup_list = cb->list;
                __wt_writeunlock(session, &conn->hot_backup_lock);
                F_SET(session, WT_SESSION_BACKUP_CURSOR);
-       } else if (cb->bfs != NULL)
+       }
+       if (cb->bfs != NULL)
                WT_TRET(__wt_fclose(session, &cb->bfs));

 done:
```
That way we won't depend on any particular implementation of `__wt_sync_and_rename()`, if it returns without closing the `cb.bfs` stream, we won't have a bug.